### PR TITLE
Support extra criu flags from the environment

### PR DESCRIPTION
**Pull request by simonis**
When testing with different `criu` versions or `criu` configurations it might be useful to be able to modify the default  `criu` command line parameters. This PR introduces the new environment variable `CRAC_CRIU_OPTS` which will be interpreted as a space separated list of `criu` flags to be appended to the hard-coded list of command line parameters. E.g.

```
$ CRAC_CRIU_OPTS="-v4 -o resume.log -W /tmp/crac" java -XX:CRaCRestoreFrom=/tmp/crac
```

This will set the logging level to 4 (thus overriding the hard-coded logging level of 1 for resuming), redirect the log to the file `resume.log` and change `criu`'s working directory (which will contain the log file) to `/tmp/crac`.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Anton Kozlov](https://openjdk.org/census#akozlov) (@AntonKozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac pull/26/head:pull/26` \
`$ git checkout pull/26`

Update a local copy of the PR: \
`$ git checkout pull/26` \
`$ git pull https://git.openjdk.org/crac pull/26/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 26`

View PR using the GUI difftool: \
`$ git pr show -t 26`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/26.diff">https://git.openjdk.org/crac/pull/26.diff</a>

</details>

### CODE REVIEW

File: src/java.base/unix/native/criuengine/criuengine.c [hunk 1]
```diff
@@ -95,15 +95,26 @@ static int checkpoint(pid_t jvm,
...
+        char *criuopts = getenv("CRAC_CRIU_OPTS");
+        if (criuopts) {
+            char* criuopt = strtok(criuopts, " ");
+            while (criuopt) {
+                *arg++ = criuopt;
+                criuopt = strtok(NULL, " ");
+            }
+        }
+        *arg++ = NULL;
+        assert(ARRAY_SIZE(args) >= (size_t)(arg - args));
```
Comment (AntonKozlov) [inline src/java.base/unix/native/criuengine/criuengine.c:116]: I'm not sure if we are compiling with asserts. Could you turn this and another assert below into a runtime check and explicit failure if arguments do not fit?

**Review by AntonKozlov — CHANGES_REQUESTED** [review]

File: src/java.base/unix/native/criuengine/criuengine.c (hunk 1 above)
Comment (simonis) [reply src/java.base/unix/native/criuengine/criuengine.c]: You're right. We compile product builds with `-DNDEBUG`. I've reworked the code to drop surplus arguments from `CRAC_CRIU_OPTS` and issue a warning if this was necessary.

**Review by AntonKozlov — APPROVED** [review]
Looks good to me. Thanks!

**Pull request closed without merging**